specifyweb/frontend/js_src/lib/components/Formatters/Formatter.tsx

```diff
@@ -169,78 +237,103 @@ function Fields({
   readonly fields: GetSet<Formatter['definition']['fields'][number]['fields']>;
   readonly onDelete: (() => void) | undefined;
 }): JSX.Element {
+  const [displayFormatter, setDisplayFormatter] = React.useState(false);
   return (
-    <table
-      /*
-       * REFACTOR: replace min-w-[35rem] with a container query that replaces
-       *   table layout with list layout
-       */
-      className={`
-        grid-table min-w-[35rem] grid-cols-[min-content_auto_auto_min-content]
-        gap-2
-      `}
-    >
-      <thead>
-        <tr>
-          <th>{resourcesText.separator()}</th>
-          <th>{schemaText.field()}</th>
-          <th>{resourcesText.formatter()}</th>
-          <td />
-        </tr>
-      </thead>
-      <tbody>
-        {fields.map((field, index) => (
-          <Field
-            field={[
-              field,
-              (field): void => setFields(replaceItem(fields, index, field)),
-            ]}
-            key={index}
-            table={table}
-            onRemove={(): void => setFields(removeItem(fields, index))}
-          />
-        ))}
-        <tr>
-          <td className="col-span-3 !gap-2">
-            <Button.Success
-              onClick={(): void =>
-                setFields([
-                  ...fields,
-                  {
-                    separator: localized(' '),
-                    aggregator: undefined,
-                    formatter: undefined,
-                    fieldFormatter: undefined,
-                    field: undefined,
-                  },
-                ])
-              }
-            >
-              {resourcesText.addField()}
-            </Button.Success>
-            {typeof handleDelete === 'function' && (
-              <Button.Danger onClick={handleDelete}>
-                {resourcesText.deleteDefinition()}
-              </Button.Danger>
-            )}
-          </td>
-          <td />
-        </tr>
-      </tbody>
-    </table>
+    <>
+      {fields.length === 0 ? null : (
+        <table
+          /*
+           * REFACTOR: replace min-w-[35rem] with a container query that replaces
+           *   table layout with list layout
+           */
+          className={`
+            grid-table min-w-[35rem]
+            gap-y-2 gap-x-4
```

Comment:
(optional) try to have the gap in both directions be the same unless there is a reason to be otherwise
(if anything, if I do decide to make gaps uneven, the gap between rows is usually larger than the gap between columns as data within the row are related to each other, while data between rows is not)